test/ca/mcgill/cs/stg/jetuml/graph/TestUsageScenariosClassDiagram.java

```diff
@@ -0,0 +1,694 @@
+	private ClassRelationshipEdge depeEdge = new DependencyEdge();
+	private ClassRelationshipEdge geneEdge = new GeneralizationEdge();
+	
+	@Before
+	public void setup()
+	{
+		diagram = new ClassDiagramGraph();
+		aGraphics = new BufferedImage(256, 256, BufferedImage.TYPE_INT_RGB).createGraphics();
+		aPanel = new GraphPanel(diagram, new ToolBar(diagram));
+		aGrid = new Grid();
+		clipboard = new Clipboard();
+		aList = new SelectionList();
+		aClassNode = new ClassNode();
+		aInterfaceNode = new InterfaceNode();
+		aPackageNode = new PackageNode();
+		aNoteNode = new NoteNode();
+		aggrEdge = new AggregationEdge();
+		assoEdge = new AssociationEdge();
+		depeEdge = new DependencyEdge();
+		geneEdge = new GeneralizationEdge();
+		
+	}
+	
+	@Test
+	public void testBasicNode()
+	{
+		diagram.addNode(aClassNode, new Point2D.Double(5, 5));
+		diagram.addNode(aInterfaceNode, new Point2D.Double(44, 44));
+		diagram.addNode(aPackageNode, new Point2D.Double(87, 87));
+		diagram.addNode(aNoteNode, new Point2D.Double(134, 132));
+
+		assertEquals(4, diagram.getRootNodes().size());
+		
+		// set up the properties for the nodes
+		aClassNode.getName().setText("truck");
+		aClassNode.getMethods().setText("setDriver()");
+		aInterfaceNode.getName().setText("vehicle");
+		aInterfaceNode.getMethods().setText("getPrice()");
+		aPackageNode.setName("object");
+		aPackageNode.getContents().setText("some stuff");
+		aNoteNode.getText().setText("some text...");
+		
+		// 1.1 test nodes properties
+		assertEquals(aClassNode.getName().getText(), "truck");
+		assertEquals(aClassNode.getMethods().getText(), "setDriver()");
+		assertEquals(aInterfaceNode.getMethods().getText(), "getPrice()");
+		assertEquals(aPackageNode.getName(), "object");
+		assertEquals(aPackageNode.getContents().getText(), "some stuff");
+		assertEquals(aNoteNode.getText().getText(), "some text...");
+	}
+	
+	@Test
+	public void testEdgeCreation()
+	{
+		diagram.addNode(aClassNode, new Point2D.Double(5, 5));
+		diagram.addNode(aInterfaceNode, new Point2D.Double(44, 44));
+		diagram.addNode(aPackageNode, new Point2D.Double(87, 87));
+		diagram.addNode(aNoteNode, new Point2D.Double(134, 132));
+		
+		// both start and end points are invalid
+		diagram.addEdge(aggrEdge, new Point2D.Double(70, 70), new Point2D.Double(170, 170));
+		// one point is invalid
+		diagram.addEdge(aggrEdge, new Point2D.Double(6, 7), new Point2D.Double(170, 170));
+		assertEquals(0, diagram.getEdges().size());
+		
+		diagram.addEdge(aggrEdge, new Point2D.Double(8, 10), new Point2D.Double(45, 48));
+		diagram.addEdge(assoEdge, new Point2D.Double(47, 49), new Point2D.Double(9, 17));
+		diagram.addEdge(depeEdge, new Point2D.Double(90, 93), new Point2D.Double(44, 49));
+		assertEquals(3, diagram.getEdges().size());
+		
+		diagram.addEdge(new AssociationEdge(), new Point2D.Double(47, 49), new Point2D.Double(50, 49));
+		assertEquals(4, diagram.getEdges().size());
+		
+		// not every edge is a valid to be self-edge
```

Comment:
This should read "// not every edge is a valid self-edge"

Can self-edge checks be extracted into a separate method?